cmd/template/cluster/flag.go

```diff
@@ -105,8 +105,14 @@ func (f *flag) Validate() error {
 
 		return nil
 	}
-	if f.Domain == "" {
-		return microerror.Maskf(invalidFlagError, "--%s must not be empty", flagDomain)
+	{
+		// Validate domain.
+		switch f.Provider {
```

Comment:
Should we throw an error if an user tries to use it on `Azure`?

Reply:
Yeah, maybe it makes sense to tell the customer that it's not allowed on Azure.

Reply:
Added